ESP32-S3 Compilation fails. debug.h and interrupts.h missing

previously raised in [ESP32 Compilation fails. debug.h and interrupts.h missing](https://github.com/eModbus/eModbus/issues/393#top)
#393 but closed as author of bug wasn't in a place to help.

**Describe the bug**
A program that was otherwise working (complied, build and loaded to target) starts to produce numerous errors at compile with the most notable being debug.h and interupts.h were missing (#include statements in ESPAsyncTCP/src/ESPAsyncTCPbuffer.cpp and ESPAsyncTCP/src/SyncClient.cpp:25:10 respectivly.

At this time I had added the eModbus Library using PIO add library GUI, and as a result it added eModbus plus several syncTCP libraries plu the alternate ethernet library to my libdeps directory.  because of the duplicated Ethernet library I removed the ethernet library I was using.

In my code I just added #include <ModbusClientRTU.h>.  I have not added any other related code (other than existing Ethernet)

**Expected behavior**
No compile errors, would be even nice if it only compiled what was called by program

**To Reproduce**
Steps to reproduce the behavior:

**Example code**
Platform.ini
```

[platformio]
default_envs = Waveshare_ESP32-S3-Eth

[env]
platform = https://github.com/pioarduino/platform-espressif32/releases/download/stable/platform-espressif32.zip
framework = arduino

[env:ws_esp32_s3_matrix]
board = ws_esp32_s3_matrix
lib_deps = 
	;arduino-libraries/Ethernet@^2.0.2
	knolleary/PubSubClient@^2.8
	adafruit/Adafruit NeoPixel@^1.15.2
	miq19/eModbus@^1.7.4

[env:Waveshare_ESP32-S3-Eth]
board_dir = boards
board = Waveshare_esp32-s3-Eth-n16r8
monitor_speed = 115200
lib_deps = 
	;arduino-libraries/Ethernet@^2.0.2
	knolleary/PubSubClient@^2.8
	adafruit/Adafruit NeoPixel@^1.15.2
	miq19/eModbus@^1.7.4

```
Start of main.src
`

#include <Arduino.h>
#include <Ethernet.h>
#include <SPI.h>
#include <PubSubClient.h> 
#include <Adafruit_NeoPixel.h>
#include <ModbusClientRTU.h>


`

Part of Error reports
`
Processing Waveshare_ESP32-S3-Eth (board: Waveshare_esp32-s3-Eth-n16r8; platform: https://github.com/pioarduino/platform-espressif32/releases/download/stable/platform-espressif32.zip; framework: arduino)
---------------------------------------------------------------------------------------------------------------------------------------------------------------Verbose mode can be enabled via `-v, --verbose` option
CONFIGURATION: https://docs.platformio.org/page/boards/espressif32/Waveshare_esp32-s3-Eth-n16r8.html
PLATFORM: Espressif 32 (55.3.35) > Waveshare ESP32-S3-Eth-N16R8V (16 MB Flash Quad, 8 MB PSRAM Octal)
HARDWARE: ESP32S3 240MHz, 320KB RAM, 16MB Flash
DEBUG: Current (esp-builtin) On-board (esp-builtin) External (cmsis-dap, esp-bridge, esp-prog, iot-bus-jtag, jlink, minimodule, olimex-arm-usb-ocd, olimex-arm-usb-ocd-h, olimex-arm-usb-tiny-h, olimex-jtag-tiny, tumpa)
PACKAGES:
 - contrib-piohome @ 3.4.4
 - framework-arduinoespressif32 @ 3.3.5
 - framework-arduinoespressif32-libs @ 5.5.0+sha.9bb7aa84fe
 - tool-esptoolpy @ 5.1.0
 - tool-mklittlefs @ 3.2.0
 - toolchain-xtensa-esp-elf @ 14.2.0+20251107
*** Applied include path shortening for 331 framework paths ***
*** Path length reduced from 39150 to ~13756 characters ***
*** Estimated savings: 24825 characters ***
LDF: Library Dependency Finder -> https://bit.ly/configure-pio-ldf
LDF Modes: Finder ~ chain, Compatibility ~ soft
Found 49 compatible libraries
Scanning dependencies...
Dependency Graph
|-- PubSubClient @ 2.8.0
|-- Adafruit NeoPixel @ 1.15.2
|-- eModbus @ 1.7.4
|-- Ethernet @ 2.0.2+sha.aa23370
|-- SPI @ 3.3.5
Building in release mode
Compiling .pio\build\Waveshare_ESP32-S3-Eth\lib526\PubSubClient\PubSubClient.cpp.o
Compiling .pio\build\Waveshare_ESP32-S3-Eth\lib8fd\Adafruit NeoPixel\Adafruit_NeoPixel.cpp.o
Compiling .pio\build\Waveshare_ESP32-S3-Eth\lib8fd\Adafruit NeoPixel\Adafruit_Neopixel_RP2.cpp.o
Compiling .pio\build\Waveshare_ESP32-S3-Eth\lib8fd\Adafruit NeoPixel\esp.c.o
Compiling .pio\build\Waveshare_ESP32-S3-Eth\lib8fd\Adafruit NeoPixel\esp8266.c.o
Compiling .pio\build\Waveshare_ESP32-S3-Eth\lib8fd\Adafruit NeoPixel\kendyte_k210.c.o
Compiling .pio\build\Waveshare_ESP32-S3-Eth\lib8fd\Adafruit NeoPixel\psoc6.c.o
Compiling .pio\build\Waveshare_ESP32-S3-Eth\lib448\SPI\SPI.cpp.o
Compiling .pio\build\Waveshare_ESP32-S3-Eth\lib37b\Ethernet\Dhcp.cpp.o
Compiling .pio\build\Waveshare_ESP32-S3-Eth\lib37b\Ethernet\Dns.cpp.o
Compiling .pio\build\Waveshare_ESP32-S3-Eth\lib37b\Ethernet\Ethernet.cpp.o
Compiling .pio\build\Waveshare_ESP32-S3-Eth\lib37b\Ethernet\EthernetClient.cpp.o
Compiling .pio\build\Waveshare_ESP32-S3-Eth\lib37b\Ethernet\EthernetServer.cpp.o
Compiling .pio\build\Waveshare_ESP32-S3-Eth\lib37b\Ethernet\EthernetUdp.cpp.o
Compiling .pio\build\Waveshare_ESP32-S3-Eth\lib37b\Ethernet\socket.cpp.o
Compiling .pio\build\Waveshare_ESP32-S3-Eth\lib37b\Ethernet\utility\w5100.cpp.o
Compiling .pio\build\Waveshare_ESP32-S3-Eth\lib137\Network\NetworkClient.cpp.o
Compiling .pio\build\Waveshare_ESP32-S3-Eth\lib137\Network\NetworkEvents.cpp.o
Compiling .pio\build\Waveshare_ESP32-S3-Eth\lib137\Network\NetworkInterface.cpp.o
Compiling .pio\build\Waveshare_ESP32-S3-Eth\lib137\Network\NetworkManager.cpp.o
Compiling .pio\build\Waveshare_ESP32-S3-Eth\lib137\Network\NetworkServer.cpp.o
Compiling .pio\build\Waveshare_ESP32-S3-Eth\lib137\Network\NetworkUdp.cpp.o
Compiling .pio\build\Waveshare_ESP32-S3-Eth\lib3f1\AsyncTCP\AsyncTCP.cpp.o
Compiling .pio\build\Waveshare_ESP32-S3-Eth\libbe6\ESPAsyncTCP\AsyncPrinter.cpp.o
Compiling .pio\build\Waveshare_ESP32-S3-Eth\libbe6\ESPAsyncTCP\ESPAsyncTCP.cpp.o
Compiling .pio\build\Waveshare_ESP32-S3-Eth\libbe6\ESPAsyncTCP\ESPAsyncTCPbuffer.cpp.o
Compiling .pio\build\Waveshare_ESP32-S3-Eth\libbe6\ESPAsyncTCP\SyncClient.cpp.o
Compiling .pio\build\Waveshare_ESP32-S3-Eth\libbe6\ESPAsyncTCP\tcp_axtls.c.o
Compiling .pio\build\Waveshare_ESP32-S3-Eth\lib0d7\eModbus\CoilData.cpp.o
Compiling .pio\build\Waveshare_ESP32-S3-Eth\lib0d7\eModbus\Logging.cpp.o
.pio/libdeps/Waveshare_ESP32-S3-Eth/ESPAsyncTCP/src/ESPAsyncTCPbuffer.cpp:27:10: fatal error: debug.h: No such file or directory

***************************************************************
* Looking for debug.h dependency? Check our library registry!
*
* CLI  > platformio lib search "header:debug.h"
* Web  > https://registry.platformio.org/search?q=header:debug.h
*
***************************************************************

   27 | #include <debug.h>
      |          ^~~~~~~~~
compilation terminated.
*** [.pio\build\Waveshare_ESP32-S3-Eth\libbe6\ESPAsyncTCP\ESPAsyncTCPbuffer.cpp.o] Error 1
.pio/libdeps/Waveshare_ESP32-S3-Eth/ESPAsyncTCP/src/SyncClient.cpp:25:10: fatal error: interrupts.h: No such file or directory

********************************************************************
* Looking for interrupts.h dependency? Check our library registry!
*
* CLI  > platformio lib search "header:interrupts.h"
* Web  > https://registry.platformio.org/search?q=header:interrupts.h
*
********************************************************************

   25 | #include <interrupts.h>
      |          ^~~~~~~~~~~~~~
compilation terminated.
*** [.pio\build\Waveshare_ESP32-S3-Eth\libbe6\ESPAsyncTCP\SyncClient.cpp.o] Error 1
.pio/libdeps/Waveshare_ESP32-S3-Eth/ESPAsyncTCP/src/AsyncPrinter.cpp: In constructor 'AsyncPrinter::AsyncPrinter(AsyncClient*, size_t)':
.pio/libdeps/Waveshare_ESP32-S3-Eth/ESPAsyncTCP/src/AsyncPrinter.cpp:48:5: error: 'panic' was not declared in this scope
   48 |     panic(); //What should we do?
      |     ^~~~~
.pio/libdeps/Waveshare_ESP32-S3-Eth/ESPAsyncTCP/src/AsyncPrinter.cpp: In member function 'int AsyncPrinter::connect(IPAddress, uint16_t)':
.pio/libdeps/Waveshare_ESP32-S3-Eth/ESPAsyncTCP/src/AsyncPrinter.cpp:71:5: error: 'panic' was not declared in this scope
   71 |     panic();
      |     ^~~~~
.pio/libdeps/Waveshare_ESP32-S3-Eth/ESPAsyncTCP/src/AsyncPrinter.cpp: In member function 'int AsyncPrinter::connect(const char*, uint16_t)':
.pio/libdeps/Waveshare_ESP32-S3-Eth/ESPAsyncTCP/src/AsyncPrinter.cpp:88:5: error: 'panic' was not declared in this scope
   88 |     panic();
      |     ^~~~~
.pio/libdeps/Waveshare_ESP32-S3-Eth/ESPAsyncTCP/src/AsyncPrinter.cpp: In member function 'void AsyncPrinter::_onConnect(AsyncClient*)':
.pio/libdeps/Waveshare_ESP32-S3-Eth/ESPAsyncTCP/src/AsyncPrinter.cpp:109:5: error: 'panic' was not declared in this scope
  109 |     panic();
      |     ^~~~~
.pio/libdeps/Waveshare_ESP32-S3-Eth/ESPAsyncTCP/src/AsyncPrinter.cpp: In member function 'AsyncPrinter& AsyncPrinter::operator=(const AsyncPrinter&)':
.pio/libdeps/Waveshare_ESP32-S3-Eth/ESPAsyncTCP/src/AsyncPrinter.cpp:130:5: error: 'panic' was not declared in this scope
  130 |     panic();
      |     ^~~~~
.pio/libdeps/Waveshare_ESP32-S3-Eth/ESPAsyncTCP/src/AsyncPrinter.cpp: In member function 'size_t AsyncPrinter::_sendBuffer()':
.pio/libdeps/Waveshare_ESP32-S3-Eth/ESPAsyncTCP/src/AsyncPrinter.cpp:182:5: error: 'panic' was not declared in this scope
  182 |     panic(); // Connection should be aborted instead
      |     ^~~~~
*** [.pio\build\Waveshare_ESP32-S3-Eth\libbe6\ESPAsyncTCP\AsyncPrinter.cpp.o] Error 1
.pio/libdeps/Waveshare_ESP32-S3-Eth/ESPAsyncTCP/src/ESPAsyncTCP.cpp: In member function 'bool AsyncClient::connect(IPAddress, uint16_t)':
.pio/libdeps/Waveshare_ESP32-S3-Eth/ESPAsyncTCP/src/ESPAsyncTCP.cpp:267:33: error: invalid user-defined conversion from 'IPAddress' to 'const ip_addr_t*' {aka 
'const ip_addr*'} [-fpermissive]
  267 |   size_t err = tcp_connect(pcb, addr, port,(tcp_connected_fn)&_s_connected);
      |                                 ^~~~
In file included from C:/Users/IansH/.platformio/packages/framework-arduinoespressif32/cores/esp32/Arduino.h:198,
                 from .pio/libdeps/Waveshare_ESP32-S3-Eth/ESPAsyncTCP/src/ESPAsyncTCP.cpp:74:
C:/Users/IansH/.platformio/packages/framework-arduinoespressif32/cores/esp32/IPAddress.h:83:3: note: candidate is: 'IPAddress::operator uint32_t() const' (near match)
   83 |   operator uint32_t() const {
      |   ^~~~~~~~
C:/Users/IansH/.platformio/packages/framework-arduinoespressif32/cores/esp32/IPAddress.h:83:3: note:   no known conversion from 'uint32_t' {aka 'long unsigned 
int'} to 'const ip_addr_t*' {aka 'const ip_addr*'}
In file included from .pio/libdeps/Waveshare_ESP32-S3-Eth/ESPAsyncTCP/src/ESPAsyncTCP.cpp:79:
C:/Users/IansH/.platformio/packages/framework-arduinoespressif32-libs/esp32s3/include/lwip/lwip/src/include/lwip/tcp.h:461:69: note:   initializing argument 2 
of 'err_t tcp_connect(tcp_pcb*, const ip_addr_t*, u16_t, tcp_connected_fn)'
  461 | err_t            tcp_connect (struct tcp_pcb *pcb, const ip_addr_t *ipaddr,
      |                                                    ~~~~~~~~~~~~~~~~~^~~~~~
.pio/libdeps/Waveshare_ESP32-S3-Eth/ESPAsyncTCP/src/ESPAsyncTCP.cpp: In member function 'bool AsyncClient::connect(const char*, uint16_t)':
.pio/libdeps/Waveshare_ESP32-S3-Eth/ESPAsyncTCP/src/ESPAsyncTCP.cpp:277:39: error: invalid user-defined conversion from 'IPAddress' to 'ip_addr_t*' {aka 'ip_addr*'} [-fpermissive]
  277 |   err_t err = dns_gethostbyname(host, addr, (dns_found_callback)&_s_dns_found, this);
      |                                       ^~~~
C:/Users/IansH/.platformio/packages/framework-arduinoespressif32/cores/esp32/IPAddress.h:83:3: note: candidate is: 'IPAddress::operator uint32_t() const' (near match)
   83 |   operator uint32_t() const {
      |   ^~~~~~~~
C:/Users/IansH/.platformio/packages/framework-arduinoespressif32/cores/esp32/IPAddress.h:83:3: note:   no known conversion from 'uint32_t' {aka 'long unsigned 
int'} to 'ip_addr_t*' {aka 'ip_addr*'}
In file included from .pio/libdeps/Waveshare_ESP32-S3-Eth/ESPAsyncTCP/src/ESPAsyncTCP.cpp:81:
C:/Users/IansH/.platformio/packages/framework-arduinoespressif32-libs/esp32s3/include/lwip/lwip/src/include/lwip/dns.h:109:69: note:   initializing argument 2 
of 'err_t dns_gethostbyname(const char*, ip_addr_t*, dns_found_callback, void*)'
  109 | err_t            dns_gethostbyname(const char *hostname, ip_addr_t *addr,
      |                                                          ~~~~~~~~~~~^~~~
.pio/libdeps/Waveshare_ESP32-S3-Eth/ESPAsyncTCP/src/ESPAsyncTCP.cpp: In member function 'bool AsyncClient::operator==(const AsyncClient&)':
.pio/libdeps/Waveshare_ESP32-S3-Eth/ESPAsyncTCP/src/ESPAsyncTCP.cpp:331:75: error: no matching function for call to 'IPAddress::IPAddress(ip_addr_t&)'
  331 |   return (_pcb != NULL && other._pcb != NULL && (IPAddress(_pcb->remote_ip) ==  IPAddress(other._pcb->remote_ip)) && (_pcb->remote_port == other._pcb->remote_port));
      |                                                                           ^
C:/Users/IansH/.platformio/packages/framework-arduinoespressif32/cores/esp32/IPAddress.h:116:3: note: candidate: 'IPAddress::IPAddress(const ip_addr_t*)'      
  116 |   IPAddress(const ip_addr_t *addr);
      |   ^~~~~~~~~
C:/Users/IansH/.platformio/packages/framework-arduinoespressif32/cores/esp32/IPAddress.h:116:30: note:   no known conversion for argument 1 from 'ip_addr_t' {aka 'ip_addr'} to 'const ip_addr_t*' {aka 'const ip_addr*'}
  116 |   IPAddress(const ip_addr_t *addr);
      |             ~~~~~~~~~~~~~~~~~^~~~
C:/Users/IansH/.platformio/packages/framework-arduinoespressif32/cores/esp32/IPAddress.h:74:3: note: candidate: 'IPAddress::IPAddress(const IPAddress&)'       
   74 |   IPAddress(const IPAddress &address);
      |   ^~~~~~~~~
C:/Users/IansH/.platformio/packages/framework-arduinoespressif32/cores/esp32/IPAddress.h:74:30: note:   no known conversion for argument 1 from 'ip_addr_t' {aka 'ip_addr'} to 'const IPAddress&'
   74 |   IPAddress(const IPAddress &address);
      |             ~~~~~~~~~~~~~~~~~^~~~~~~
C:/Users/IansH/.platformio/packages/framework-arduinoespressif32/cores/esp32/IPAddress.h:73:3: note: candidate: 'IPAddress::IPAddress(const char*)'
   73 |   IPAddress(const char *address);
      |   ^~~~~~~~~
C:/Users/IansH/.platformio/packages/framework-arduinoespressif32/cores/esp32/IPAddress.h:73:25: note:   no known conversion for argument 1 from 'ip_addr_t' {aka 'ip_addr'} to 'const char*'
   73 |   IPAddress(const char *address);
      |             ~~~~~~~~~~~~^~~~~~~
C:/Users/IansH/.platformio/packages/framework-arduinoespressif32/cores/esp32/IPAddress.h:71:3: note: candidate: 'IPAddress::IPAddress(IPType, const uint8_t*, uint8_t)'
   71 |   IPAddress(IPType ip_type, const uint8_t *address, uint8_t z = 0);
      |   ^~~~~~~~~
C:/Users/IansH/.platformio/packages/framework-arduinoespressif32/cores/esp32/IPAddress.h:71:3: note:   candidate expects 3 arguments, 1 provided
C:/Users/IansH/.platformio/packages/framework-arduinoespressif32/cores/esp32/IPAddress.h:70:3: note: candidate: 'IPAddress::IPAddress(const uint8_t*)'
   70 |   IPAddress(const uint8_t *address);
      |   ^~~~~~~~~
C:/Users/IansH/.platformio/packages/framework-arduinoespressif32/cores/esp32/IPAddress.h:70:28: note:   no known conversion for argument 1 from 'ip_addr_t' {aka 'ip_addr'} to 'const uint8_t*' {aka 'const unsigned char*'}
   70 |   IPAddress(const uint8_t *address);
      |             ~~~~~~~~~~~~~~~^~~~~~~
C:/Users/IansH/.platformio/packages/framework-arduinoespressif32/cores/esp32/IPAddress.h:68:3: note: candidate: 'IPAddress::IPAddress(uint32_t)'
   68 |   IPAddress(uint32_t address);
      |   ^~~~~~~~~
C:/Users/IansH/.platformio/packages/framework-arduinoespressif32/cores/esp32/IPAddress.h:68:22: note:   no known conversion for argument 1 from 'ip_addr_t' {aka 'ip_addr'} to 'uint32_t' {aka 'long unsigned int'}
   68 |   IPAddress(uint32_t address);
      |             ~~~~~~~~~^~~~~~~
C:/Users/IansH/.platformio/packages/framework-arduinoespressif32/cores/esp32/IPAddress.h:63:3: note: candidate: 'IPAddress::IPAddress(uint8_t, uint8_t, uint8_t, uint8_t, uint8_t, uint8_t, uint8_t, uint8_t, uint8_t, uint8_t, uint8_t, uint8_t, uint8_t, uint8_t, uint8_t, uint8_t, uint8_t)'
   63 |   IPAddress(
      |   ^~~~~~~~~
C:/Users/IansH/.platformio/packages/framework-arduinoespressif32/cores/esp32/IPAddress.h:63:3: note:   candidate expects 17 arguments, 1 provided
C:/Users/IansH/.platformio/packages/framework-arduinoespressif32/cores/esp32/IPAddress.h:62:3: note: candidate: 'IPAddress::IPAddress(uint8_t, uint8_t, uint8_t, uint8_t)'
   62 |   IPAddress(uint8_t first_octet, uint8_t second_octet, uint8_t third_octet, uint8_t fourth_octet);
      |   ^~~~~~~~~
C:/Users/IansH/.platformio/packages/framework-arduinoespressif32/cores/esp32/IPAddress.h:62:3: note:   candidate expects 4 arguments, 1 provided
C:/Users/IansH/.platformio/packages/framework-arduinoespressif32/cores/esp32/IPAddress.h:61:3: note: candidate: 'IPAddress::IPAddress(IPType)'
   61 |   IPAddress(IPType ip_type);
      |   ^~~~~~~~~
C:/Users/IansH/.platformio/packages/framework-arduinoespressif32/cores/esp32/IPAddress.h:61:20: note:   no known conversion for argument 1 from 'ip_addr_t' {aka 'ip_addr'} to 'IPType'
   61 |   IPAddress(IPType ip_type);
      |             ~~~~~~~^~~~~~~
C:/Users/IansH/.platformio/packages/framework-arduinoespressif32/cores/esp32/IPAddress.h:60:3: note: candidate: 'IPAddress::IPAddress()'
   60 |   IPAddress();
      |   ^~~~~~~~~
C:/Users/IansH/.platformio/packages/framework-arduinoespressif32/cores/esp32/IPAddress.h:60:3: note:   candidate expects 0 arguments, 1 provided
.pio/libdeps/Waveshare_ESP32-S3-Eth/ESPAsyncTCP/src/ESPAsyncTCP.cpp:331:112: error: no matching function for call to 'IPAddress::IPAddress(ip_addr_t&)'        
  331 |   return (_pcb != NULL && other._pcb != NULL && (IPAddress(_pcb->remote_ip) ==  IPAddress(other._pcb->remote_ip)) && (_pcb->remote_port == other._pcb->remote_port));
      |                                                                                                                ^
C:/Users/IansH/.platformio/packages/framework-arduinoespressif32/cores/esp32/IPAddress.h:116:3: note: candidate: 'IPAddress::IPAddress(const ip_addr_t*)'      
  116 |   IPAddress(const ip_addr_t *addr);
      |   ^~~~~~~~~
C:/Users/IansH/.platformio/packages/framework-arduinoespressif32/cores/esp32/IPAddress.h:116:30: note:   no known conversion for argument 1 from 'ip_addr_t' {aka 'ip_addr'} to 'const ip_addr_t*' {aka 'const ip_addr*'}
  116 |   IPAddress(const ip_addr_t *addr);
      |             ~~~~~~~~~~~~~~~~~^~~~
C:/Users/IansH/.platformio/packages/framework-arduinoespressif32/cores/esp32/IPAddress.h:74:3: note: candidate: 'IPAddress::IPAddress(const IPAddress&)'       
   74 |   IPAddress(const IPAddress &address);
      |   ^~~~~~~~~
C:/Users/IansH/.platformio/packages/framework-arduinoespressif32/cores/esp32/IPAddress.h:74:30: note:   no known conversion for argument 1 from 'ip_addr_t' {aka 'ip_addr'} to 'const IPAddress&'
   74 |   IPAddress(const IPAddress &address);
      |             ~~~~~~~~~~~~~~~~~^~~~~~~
C:/Users/IansH/.platformio/packages/framework-arduinoespressif32/cores/esp32/IPAddress.h:73:3: note: candidate: 'IPAddress::IPAddress(const char*)'
   73 |   IPAddress(const char *address);
      |   ^~~~~~~~~
C:/Users/IansH/.platformio/packages/framework-arduinoespressif32/cores/esp32/IPAddress.h:73:25: note:   no known conversion for argument 1 from 'ip_addr_t' {aka 'ip_addr'} to 'const char*'
   73 |   IPAddress(const char *address);
      |             ~~~~~~~~~~~~^~~~~~~
C:/Users/IansH/.platformio/packages/framework-arduinoespressif32/cores/esp32/IPAddress.h:71:3: note: candidate: 'IPAddress::IPAddress(IPType, const uint8_t*, uint8_t)'
   71 |   IPAddress(IPType ip_type, const uint8_t *address, uint8_t z = 0);
      |   ^~~~~~~~~
C:/Users/IansH/.platformio/packages/framework-arduinoespressif32/cores/esp32/IPAddress.h:71:3: note:   candidate expects 3 arguments, 1 provided
C:/Users/IansH/.platformio/packages/framework-arduinoespressif32/cores/esp32/IPAddress.h:70:3: note: candidate: 'IPAddress::IPAddress(const uint8_t*)'
   70 |   IPAddress(const uint8_t *address);
      |   ^~~~~~~~~
C:/Users/IansH/.platformio/packages/framework-arduinoespressif32/cores/esp32/IPAddress.h:70:28: note:   no known conversion for argument 1 from 'ip_addr_t' {aka 'ip_addr'} to 'const uint8_t*' {aka 'const unsigned char*'}
   70 |   IPAddress(const uint8_t *address);
      |             ~~~~~~~~~~~~~~~^~~~~~~
C:/Users/IansH/.platformio/packages/framework-arduinoespressif32/cores/esp32/IPAddress.h:68:3: note: candidate: 'IPAddress::IPAddress(uint32_t)'
   68 |   IPAddress(uint32_t address);
      |   ^~~~~~~~~
C:/Users/IansH/.platformio/packages/framework-arduinoespressif32/cores/esp32/IPAddress.h:68:22: note:   no known conversion for argument 1 from 'ip_addr_t' {aka 'ip_addr'} to 'uint32_t' {aka 'long unsigned int'}
   68 |   IPAddress(uint32_t address);
      |             ~~~~~~~~~^~~~~~~
C:/Users/IansH/.platformio/packages/framework-arduinoespressif32/cores/esp32/IPAddress.h:63:3: note: candidate: 'IPAddress::IPAddress(uint8_t, uint8_t, uint8_t, uint8_t, uint8_t, uint8_t, uint8_t, uint8_t, uint8_t, uint8_t, uint8_t, uint8_t, uint8_t, uint8_t, uint8_t, uint8_t, uint8_t)'
   63 |   IPAddress(
      |   ^~~~~~~~~
C:/Users/IansH/.platformio/packages/framework-arduinoespressif32/cores/esp32/IPAddress.h:63:3: note:   candidate expects 17 arguments, 1 provided
C:/Users/IansH/.platformio/packages/framework-arduinoespressif32/cores/esp32/IPAddress.h:62:3: note: candidate: 'IPAddress::IPAddress(uint8_t, uint8_t, uint8_t, uint8_t)'
   62 |   IPAddress(uint8_t first_octet, uint8_t second_octet, uint8_t third_octet, uint8_t fourth_octet);
      |   ^~~~~~~~~
C:/Users/IansH/.platformio/packages/framework-arduinoespressif32/cores/esp32/IPAddress.h:62:3: note:   candidate expects 4 arguments, 1 provided
C:/Users/IansH/.platformio/packages/framework-arduinoespressif32/cores/esp32/IPAddress.h:61:3: note: candidate: 'IPAddress::IPAddress(IPType)'
   61 |   IPAddress(IPType ip_type);
      |   ^~~~~~~~~
C:/Users/IansH/.platformio/packages/framework-arduinoespressif32/cores/esp32/IPAddress.h:61:20: note:   no known conversion for argument 1 from 'ip_addr_t' {aka 'ip_addr'} to 'IPType'
   61 |   IPAddress(IPType ip_type);
      |             ~~~~~~~^~~~~~~
C:/Users/IansH/.platformio/packages/framework-arduinoespressif32/cores/esp32/IPAddress.h:60:3: note: candidate: 'IPAddress::IPAddress()'
   60 |   IPAddress();
      |   ^~~~~~~~~
C:/Users/IansH/.platformio/packages/framework-arduinoespressif32/cores/esp32/IPAddress.h:60:3: note:   candidate expects 0 arguments, 1 provided
.pio/libdeps/Waveshare_ESP32-S3-Eth/ESPAsyncTCP/src/ESPAsyncTCP.cpp: In member function 'IPAddress AsyncClient::remoteIP()':
.pio/libdeps/Waveshare_ESP32-S3-Eth/ESPAsyncTCP/src/ESPAsyncTCP.cpp:861:23: error: call of overloaded 'IPAddress(int)' is ambiguous
  861 |     return IPAddress(0);
      |                       ^
C:/Users/IansH/.platformio/packages/framework-arduinoespressif32/cores/esp32/IPAddress.h:116:3: note: candidate: 'IPAddress::IPAddress(const ip_addr_t*)'      
  116 |   IPAddress(const ip_addr_t *addr);
      |   ^~~~~~~~~
C:/Users/IansH/.platformio/packages/framework-arduinoespressif32/cores/esp32/IPAddress.h:74:3: note: candidate: 'IPAddress::IPAddress(const IPAddress&)'       
   74 |   IPAddress(const IPAddress &address);
      |   ^~~~~~~~~
C:/Users/IansH/.platformio/packages/framework-arduinoespressif32/cores/esp32/IPAddress.h:73:3: note: candidate: 'IPAddress::IPAddress(const char*)'
   73 |   IPAddress(const char *address);
      |   ^~~~~~~~~
C:/Users/IansH/.platformio/packages/framework-arduinoespressif32/cores/esp32/IPAddress.h:70:3: note: candidate: 'IPAddress::IPAddress(const uint8_t*)'
   70 |   IPAddress(const uint8_t *address);
      |   ^~~~~~~~~
C:/Users/IansH/.platformio/packages/framework-arduinoespressif32/cores/esp32/IPAddress.h:68:3: note: candidate: 'IPAddress::IPAddress(uint32_t)'
   68 |   IPAddress(uint32_t address);
      |   ^~~~~~~~~
.pio/libdeps/Waveshare_ESP32-S3-Eth/ESPAsyncTCP/src/ESPAsyncTCP.cpp:862:16: error: could not convert '((AsyncClient*)this)->AsyncClient::_pcb->tcp_pcb::remote_ip' from 'ip_addr_t' {aka 'ip_addr'} to 'IPAddress'
  862 |   return _pcb->remote_ip;
      |          ~~~~~~^~~~~~~~~
      |                |
      |                ip_addr_t {aka ip_addr}
.pio/libdeps/Waveshare_ESP32-S3-Eth/ESPAsyncTCP/src/ESPAsyncTCP.cpp: In member function 'IPAddress AsyncClient::localIP()':
.pio/libdeps/Waveshare_ESP32-S3-Eth/ESPAsyncTCP/src/ESPAsyncTCP.cpp:871:23: error: call of overloaded 'IPAddress(int)' is ambiguous
  871 |     return IPAddress(0);
      |                       ^
C:/Users/IansH/.platformio/packages/framework-arduinoespressif32/cores/esp32/IPAddress.h:116:3: note: candidate: 'IPAddress::IPAddress(const ip_addr_t*)'      
  116 |   IPAddress(const ip_addr_t *addr);
      |   ^~~~~~~~~
C:/Users/IansH/.platformio/packages/framework-arduinoespressif32/cores/esp32/IPAddress.h:74:3: note: candidate: 'IPAddress::IPAddress(const IPAddress&)'
   74 |   IPAddress(const IPAddress &address);
      |   ^~~~~~~~~
C:/Users/IansH/.platformio/packages/framework-arduinoespressif32/cores/esp32/IPAddress.h:73:3: note: candidate: 'IPAddress::IPAddress(const char*)'
   73 |   IPAddress(const char *address);
      |   ^~~~~~~~~
C:/Users/IansH/.platformio/packages/framework-arduinoespressif32/cores/esp32/IPAddress.h:70:3: note: candidate: 'IPAddress::IPAddress(const uint8_t*)'
   70 |   IPAddress(const uint8_t *address);
      |   ^~~~~~~~~
C:/Users/IansH/.platformio/packages/framework-arduinoespressif32/cores/esp32/IPAddress.h:68:3: note: candidate: 'IPAddress::IPAddress(uint32_t)'
   68 |   IPAddress(uint32_t address);
      |   ^~~~~~~~~
.pio/libdeps/Waveshare_ESP32-S3-Eth/ESPAsyncTCP/src/ESPAsyncTCP.cpp:872:16: error: could not convert '((AsyncClient*)this)->AsyncClient::_pcb->tcp_pcb::local_ip' from 'ip_addr_t' {aka 'ip_addr'} to 'IPAddress'
  872 |   return _pcb->local_ip;
      |          ~~~~~~^~~~~~~~
      |                |
      |                ip_addr_t {aka ip_addr}
.pio/libdeps/Waveshare_ESP32-S3-Eth/ESPAsyncTCP/src/ESPAsyncTCP.cpp: In member function 'void AsyncServer::begin()':
.pio/libdeps/Waveshare_ESP32-S3-Eth/ESPAsyncTCP/src/ESPAsyncTCP.cpp:1128:23: error: invalid user-defined conversion from 'IPAddress' to 'const ip_addr_t*' {aka 'const ip_addr*'} [-fpermissive]
 1128 |   err = tcp_bind(pcb, local_addr, _port);
      |                       ^~~~~~~~~~
C:/Users/IansH/.platformio/packages/framework-arduinoespressif32/cores/esp32/IPAddress.h:83:3: note: candidate is: 'IPAddress::operator uint32_t() const' (near match)
   83 |   operator uint32_t() const {
      |   ^~~~~~~~
C:/Users/IansH/.platformio/packages/framework-arduinoespressif32/cores/esp32/IPAddress.h:83:3: note:   no known conversion from 'uint32_t' {aka 'long unsigned 
int'} to 'const ip_addr_t*' {aka 'const ip_addr*'}
C:/Users/IansH/.platformio/packages/framework-arduinoespressif32-libs/esp32s3/include/lwip/lwip/src/include/lwip/tcp.h:458:69: note:   initializing argument 2 
of 'err_t tcp_bind(tcp_pcb*, const ip_addr_t*, u16_t)'
  458 | err_t            tcp_bind    (struct tcp_pcb *pcb, const ip_addr_t *ipaddr,
      |                                                    ~~~~~~~~~~~~~~~~~^~~~~~
*** [.pio\build\Waveshare_ESP32-S3-Eth\libbe6\ESPAsyncTCP\ESPAsyncTCP.cpp.o] Error 1
================================================================= [FAILED] Took 39.15 seconds =================================================================
Environment             Status    Duration
----------------------  --------  ------------
Waveshare_ESP32-S3-Eth  FAILED    00:00:39.147
============================================================ 1 failed, 0 succeeded in 00:00:39.147 ============================================================
`

**Additional context**
IDE, 
VSC +PIOArduino on W10 PC  all up to date
Board 
Waveshare ESP32-S3-Eth

--- a/.github/workflows/build_platformio.yml
+++ b/.github/workflows/build_platformio.yml
@@ -30,9 +30,9 @@ jobs:
       - name: Install PlatformIO Core
         run: pip install --upgrade platformio
       - name: Download external libraries
-        run: pio pkg install --global --library https://github.com/arduino-libraries/Ethernet.git --library ESP32Async/AsyncTCP
+        run: pio pkg install --global --library https://github.com/arduino-libraries/Ethernet.git --library ESP32Async/AsyncTCP --library ESP32Async/ESPAsyncTCP
       - name: Build PlatformIO examples
-        run: pio ci --lib="." --project-option="lib_ldf_mode=deep+" --board=lolin32
+        run: pio ci --lib="." --project-option="lib_ldf_mode=deep+" --project-option="lib_compat_mode=strict" --board=lolin32
         env:
           PLATFORMIO_CI_SRC: ${{ matrix.example }}
 
@@ -63,9 +63,9 @@ jobs:
       - name: Install PlatformIO Core
         run: pip install --upgrade platformio
       - name: Download external libraries
-        run: pio pkg install --global --library https://github.com/arduino-libraries/Ethernet.git --library ESP32Async/AsyncTCP
+        run: pio pkg install --global --library https://github.com/arduino-libraries/Ethernet.git --library ESP32Async/AsyncTCP --library ESP32Async/ESPAsyncTCP
       - name: Build PlatformIO examples
-        run: pio ci --lib="." --project-option="lib_ldf_mode=deep+" --board=lolin32
+        run: pio ci --lib="." --project-option="lib_ldf_mode=deep+" --project-option="lib_compat_mode=strict" --board=lolin32
         env:
           PLATFORMIO_CI_SRC: ${{ matrix.example }}
 

--- a/library.json
+++ b/library.json
@@ -64,6 +64,7 @@
   ],
   "build":
   {
-    "lib_ldf_Mode": "deep+"
+    "libLDFMode": "deep+",
+    "libCompatMode": "strict"
   }
 }

